models/query/report-query.js

```diff
@@ -51,8 +51,11 @@ ReportQuery.prototype.toMongooseFilter = function() {
   filter = _.omitBy(filter, _.isNil);
   if (this.before)    filter.storedAt = { $lte: this.before }
   if (this.after)     filter.storedAt = Object.assign({}, filter.storedAt, { $gte: this.after });
-  if (this.author)    filter.author = { $regex: this.author, $options: 'i'}
-  if (this.keywords)  filter.content = { $regex: this.keywords,  $options: 'i' }
+  //Two step search for content/author. First search for any terms in content or author using the indexed $text search.
+  //Second step is to match exact phrase using regex in the returned superset of the documents from first step.
+  if (this.author || this.keywords) filter.$and = [{$text: { $search: this.author +" "+ this.keywords}}];
```

Comment:
Note: `null + " " + undefined === "null undefined"` so this could unexpectedly narrow results unless we know author and keywords must be empty strings like `""`. To be safer can we do:

```js
{ $search: `${this.author || ""} ${this.keywords || ""}` }
```